### ISSUE TITLE
Improving STARsolo feature statistics summary output (Gene.stats)

### ISSUE THREAD

**Issue by akaever** [alexdobin/STAR#745]
Thanks for your great work on STARsolo. Unfortunately, parsing the Feature/Gene.stats output file and calculating some basic QC stats does not seem to be so straightforward (maybe I missed some other output file):

1. It would be very helpful to have the initial total number of reads as top readout in this file. Currently, one has to add `nNinBarcode`, `nUMIhomopolymer`, `nTooMany`, `nNoMatch`, `nNoExactMatch`, `nUnmapped`, `nNoFeature`, `nAmbigFeature`, `nMatch` (`nNoFeature` missing in the doc)
1. There is at least `nTooMany` found in both sections `Barcodes:` and `Gene:`. Would it make sense to drop these section headers and use unique readout names to simplify parsing?
1. It looks like the number of exact barcode matching reads is reported only after mapping to features/genes (`nExactMatch`). For clean read demultiplexing stats it would be nice to have this number before any feature mapping.
1. As far as I have seen, there is no sum of demultiplexed reads per barcode ("barcode library size") output file. This would be very helpful, e.g. when comparing the sum of deduplicated UMIs vs. the number of demultiplexed reads per barcode.

**Comment by alexdobin** [comment]
Hi @akaever 

these are good suggestions, thanks! This outputs statistics for STARsolo are not fixed and will be changing with the change in algorithms, user requests, etc. Also, this is intended for more advanced usage, e.g. debugging of the problems. If you try [soloDevelop](https://github.com/alexdobin/STAR/tree/soloDevelop) branch, it outputs Summary.csv file that is similar to CellRangers's summary output and is more useful for normal QC of the libraries.
This branch also rearranges the features stats.

Cheers
Alex
